contentcuration/contentcuration/frontend/shared/views/channel/__tests__/channelDetailsModal.spec.js

```diff
@@ -1,99 +1,236 @@
+      expect(screen.getByText(testChannel.name)).toBeInTheDocument();
+    });
+
+    const closeButton = screen.getByRole('button', { name: /close/i });
+    await user.click(closeButton);
+
+    await waitFor(() => {
+      expect(screen.queryByTestId('modal-wrapper')).not.toBeInTheDocument();
+    });
   });
 
-  it('clicking download CSV button should call generateChannelsCSV', async () => {
-    await wrapper.setData({ loading: false });
-    const generateChannelsCSV = jest.spyOn(wrapper.vm, 'generateChannelsCSV');
-    generateChannelsCSV.mockImplementation(() => Promise.resolve());
-    await wrapper.findComponent('[data-test="dl-csv"]').trigger('click');
-    expect(generateChannelsCSV).toHaveBeenCalledWith([wrapper.vm.channelWithDetails]);
+  it('should close modal when ESC key is pressed', async () => {
+    const user = userEvent.setup();
+    renderComponent();
+
+    await waitFor(() => {
+      expect(screen.getByTestId('modal-wrapper')).toBeInTheDocument();
+    });
+
+    await user.keyboard('{Escape}');
+
+    await waitFor(() => {
+      expect(screen.queryByTestId('modal-wrapper')).not.toBeInTheDocument();
+    });
   });
 
-  describe('load', () => {
-    let loadChannel;
-    let loadChannelDetails;
+  it('should display download button after loading', async () => {
+    renderComponent();
 
-    beforeEach(() => {
-      loadChannel = jest.spyOn(wrapper.vm, 'loadChannel');
-      loadChannelDetails = jest.spyOn(wrapper.vm, 'loadChannelDetails');
+    await waitFor(() => {
+      expect(screen.getByText('Download channel summary')).toBeInTheDocument();
     });
+  });
+
+  it('should display details panel after loading', async () => {
+    renderComponent();
 
-    it('should automatically close if loadChannel does not find a channel', async () => {
-      await wrapper.vm.load();
-      expect(wrapper.vm.dialog).toBe(false);
+    await waitFor(() => {
+      expect(screen.getByTestId('details-panel')).toBeInTheDocument();
     });
+  });
 
-    it('load should call loadChannel and loadChannelDetails', async () => {
-      await wrapper.vm.load();
-      expect(loadChannel).toHaveBeenCalled();
-      expect(loadChannelDetails).toHaveBeenCalled();
+  it('should call loadChannel and loadChannelDetails on mount', async () => {
+    renderComponent();
+
+    await waitFor(() => {
+      expect(mockActions.loadChannel).toHaveBeenCalledWith(expect.any(Object), channelId);
+      expect(mockActions.loadChannelDetails).toHaveBeenCalledWith(expect.any(Object), channelId);
     });
+  });
+
+  it('should track analytics on mount', async () => {
+    renderComponent();
 
-    it('load should update document.title', async () => {
-      const channel = { name: 'testing channel' };
-      loadChannel.mockImplementation(() => Promise.resolve(channel));
-      await wrapper.vm.load();
-      expect(document.title).toContain(channel.name);
+    await waitFor(() => {
+      expect(screen.getByText(testChannel.name)).toBeInTheDocument();
```

Comment:
You attempt to test that `'should track analytics on mount'` but this expect doesn't check anything like that. I think it'd be fine to not test analytics though. You can either fix, or remove.